From 8636b419bcccf44fe85d79ad416a5a057b256b53 Mon Sep 17 00:00:00 2001
From: Henry Yen <henry.yen@mediatek.com>
Date: Fri, 19 Jan 2024 11:11:19 +0800
Subject: [PATCH 013/120] mtk: wifi: mt76: mt7996: initialize variable to avoid
 unexpected IRQ handling

Initialize the variable to avoid processing unexpected interrupts given
from wrong source.

CR-Id: WCNCR00185554
Change-Id: I4ba7465f84f253035261a8b07951add9ca5b2f6a
Signed-off-by: Henry Yen <henry.yen@mediatek.com>
Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/mmio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7996/mmio.c b/mt7996/mmio.c
index fd4d41d90..367a204d2 100644
--- a/mt7996/mmio.c
+++ b/mt7996/mmio.c
@@ -519,7 +519,7 @@ static void mt7996_irq_tasklet(struct tasklet_struct *t)
 	struct mt7996_dev *dev = from_tasklet(dev, t, mt76.irq_tasklet);
 	struct mtk_wed_device *wed = &dev->mt76.mmio.wed;
 	struct mtk_wed_device *wed_hif2 = &dev->mt76.mmio.wed_hif2;
-	u32 i, intr, mask, intr1;
+	u32 i, intr, mask, intr1 = 0;
 
 	if (dev->hif2 && mtk_wed_device_active(wed_hif2)) {
 		mtk_wed_device_irq_set_mask(wed_hif2, 0);
-- 
2.39.2

